sys-apps/texinfo/texinfo-7.0.3.recipe

```diff
@@ -1,52 +1,59 @@
 SUMMARY="Standard GNU documentation format tool"
-DESCRIPTION="
-Texinfo is the official documentation format of the GNU project."
+DESCRIPTION="Texinfo is the official documentation format of the GNU project."
 HOMEPAGE="http://www.gnu.org/software/texinfo/"
 COPYRIGHT="1992-2008 Free Software Foundation, Inc."
 LICENSE="GNU GPL v3"
-REVISION="2"
+REVISION="3"
 SOURCE_URI="http://ftp.gnu.org/gnu/texinfo/texinfo-$portVersion.tar.gz"
 CHECKSUM_SHA256="3cc5706fb086b895e1dc2b407aade9f95a3a233ff856273e2b659b089f117683"
 PATCHES="texinfo-$portVersion.patchset"
 
 ARCHITECTURES="all !x86_gcc2"
+SECONDARY_ARCHITECTURES="x86"
 
 PROVIDES="
-	texinfo = $portVersion compat >= 4
-	cmd:info = $portVersion compat >= 4
-	cmd:infokey = $portVersion compat >= 4
-	cmd:install_info = $portVersion compat >= 4
-	cmd:makeinfo = $portVersion compat >= 4
-	cmd:pdftexi2dvi = $portVersion compat >= 4
-	cmd:pod2texi = $portVersion compat >= 6
-	cmd:texi2any = $portVersion compat >= 6
-	cmd:texi2dvi = $portVersion compat >= 4
-	cmd:texi2pdf = $portVersion compat >= 4
-	cmd:texindex = $portVersion compat >= 4
+	texinfo$secondaryArchSuffix = $portVersion compat >= 4
+	cmd:info$secondaryArchSuffix = $portVersion compat >= 4
+	cmd:infokey$secondaryArchSuffix = $portVersion compat >= 4
+	cmd:install_info$secondaryArchSuffix = $portVersion compat >= 4
+	cmd:makeinfo$secondaryArchSuffix = $portVersion compat >= 4
+	cmd:pdftexi2dvi$secondaryArchSuffix = $portVersion compat >= 4
+	cmd:pod2texi$secondaryArchSuffix = $portVersion compat >= 6
+	cmd:texi2any$secondaryArchSuffix = $portVersion compat >= 6
+	cmd:texi2dvi$secondaryArchSuffix = $portVersion compat >= 4
+	cmd:texi2pdf$secondaryArchSuffix = $portVersion compat >= 4
+	cmd:texindex$secondaryArchSuffix = $portVersion compat >= 4
 	"
 REQUIRES="
-	haiku
+	haiku$secondaryArchSuffix
 	cmd:gawk
-	lib:libiconv
-	lib:libintl
-	lib:libncurses
-	lib:libperl
+	lib:libiconv$secondaryArchSuffix
+	lib:libintl$secondaryArchSuffix
+	lib:libncurses$secondaryArchSuffix
+	lib:libperl$secondaryArchSuffix
 	"
+if [ -n "$secondaryArchSuffix" ]; then
+CONFLICTS="
+	texinfo
+	"
+fi
```

Comment:
As discussed on IRC with @Begasus:

The conflict is problematic because at least `automake` and `autoconf` require the primary arch texinfo.

How should we solve this?
 * ignore it and hope for the best? (not good)
 * use something like `--datadir=$dataDir$secondaryArchSubDir`, putting the data into `data/x86`?

The documentation can be removed from the secondary package.

Reply:
As mentioned there, I'm not a big fan on using a $secondaryArchSubDir in $dataDir (afaik it should be arch independent).
Checked rebuilds for autoconf and automake without the cycling dependency for texinfo (it wasn't there before so can be safely removed), rebuilds are fine and the conflict will only remove the primary texinfo when installing the secondary one.
My 2 cents for now, keep the conflict and let's see what others will say here.

Reply:
Using a different dataDir as mentioned above wouldn't work anyway because the locale files would end up wrong.

Another alternative would be to use a different package name in autoconf, so that the files end up in a different directory. It would be a bit hacky though.

Reply:
I resolved the conflict by renaming the package (in autoconf) as described before and removing the documentation from the secondary package.

If this is not wanted and the conflict should be kept instead, I can also revert the last commit again.

Reply:
I reverted the hack. We will keep the conflict, now that autoconf (#9223) and automake (#9224) are fixed.